tests/integration/targets/mso_user/tasks/main.yml

```diff
@@ -116,6 +130,121 @@
     - nm_add_user_again.current.id == nm_add_user.current.id
     - nm_add_user_again.current.username == 'ansible_test'
 
+- name: Add user in read only mode mode (check_mode)
+  mso_user: &user_present2
+    host: '{{ mso_hostname }}'
+    username: '{{ mso_username }}'
+    password: '{{ mso_password }}'
+    validate_certs: '{{ mso_validate_certs | default(false) }}'
+    use_ssl: '{{ mso_use_ssl | default(true) }}'
+    use_proxy: '{{ mso_use_proxy | default(true) }}'
+    output_level: '{{ mso_output_level | default("info") }}'
+    user: ansible_test_read
+    user_password: '#123455#123455Aa'
+    # NOTE: First name, last name, phone and email are mandatory on creation
+    first_name: Ansible2
+    last_name: Test2
+    email: mso3@cisco.com
+    phone: +32 478 436 299
+    account_status: active
+    roles:
+    - name: powerUser
+      access_type: read
+    domain: Local
+    state: present
+  check_mode: yes
+  register: cm_add_user2
+
+- name: Verify cm_add_user2
+  assert:
+    that:
+    - cm_add_user2 is changed
+    - cm_add_user2.previous == {}
+    - cm_add_user2.current.id is not defined
+    - cm_add_user2.current.username == 'ansible_test_read'
+    - cm_add_user2.current.lastName == 'Test2'
+    - cm_add_user2.current.firstName == 'Ansible2'
+    - cm_add_user2.current.emailAddress == 'mso3@cisco.com'
+    - cm_add_user2.current.phoneNumber == '+32 478 436 299'
+    - cm_add_user2.current.accountStatus == 'active'
+    - cm_add_user2.current.roles[0].accessType == 'readOnly'
+
+- name: Add user in read only (normal mode)
+  mso_user: *user_present2
+  register: nm_add_user2
+
+- name: Verify nm_add_user2
+  assert:
+    that:
+    - nm_add_user2 is changed
+    - nm_add_user2.current.id is defined
+    - nm_add_user2.current.username == 'ansible_test_read'
+    - nm_add_user2.current.lastName == 'Test2'
+    - nm_add_user2.current.firstName == 'Ansible2'
+    - nm_add_user2.current.emailAddress == 'mso3@cisco.com'
+    - nm_add_user2.current.phoneNumber == '+32 478 436 299'
+    - nm_add_user2.current.accountStatus == 'active'
+    - nm_add_user2.current.roles[0].accessType == 'readOnly'
+
+- name: Add user in read only again(check mode)
+  mso_user: 
+    <<: *user_present2
+    user_password: '#123455#123455Aa'
+  check_mode: yes  
+  register: cm_add_user2_again
+
+- name: Add user in read only again (normal mode)
```

Comment:
"in read only again" -> "with read only role again"

Reply:
Seems to still be here

Reply:
Weird. That line doesn't even exist on my branch

Reply:
It is still here so we need to see how we can change that.